codegen/datamodelgen.py

```diff
@@ -134,6 +140,29 @@ def _populate_static_info(self):
                     info.static_info = self._get_static_info(info.rules, session)
             session.exit()
 
+        if run_icing_mode:
+            session = pyfluent.launch_fluent(mode="solver-icing")
+            for _, info in self._static_info.items():
+                if info.mode == "flicing":
+                    info.static_info = self._get_static_info(info.rules, session)
+                    try:
+                        if (
+                            len(
+                                info.static_info.singletons["Case"]
+                                .singletons["App"]
+                                .singletons
+                            )
+                            == 0
+                        ):
+                            print(
+                                "Information: Icing settings not generated ( R23.1+ is required )\n"
```

Comment:
If v222 is used, this printout will show  (only if PYFLUENT_FLUENT_ROOT was defined) , as the icing datamodel is not available.